zerver/lib/streams.py

```diff
@@ -602,7 +602,11 @@ def list_to_streams(
     else:
         # autocreate=True path starts here
         if not user_profile.can_create_streams():
-            raise JsonableError(_("User cannot create streams."))
+            if user_profile.realm.create_stream_policy == Realm.POLICY_ADMINS_ONLY:
+                raise JsonableError(_("Only administrators can create streams."))
+            if user_profile.realm.create_stream_policy == Realm.POLICY_FULL_MEMBERS_ONLY:
+                raise JsonableError(_("Your account is too new to create streams."))
+            raise JsonableError(_("Not allowed for guest users"))
         elif not autocreate:
```

Comment:
I know that this error is not correct, but I have some doubt. When we directly call to the api, `add_subscriptions_backend` is called, the guest case is handled by a decorator we use on this, so the error on decorator is shown. But in one place in tests, essentially [here](https://github.com/zulip/zulip/blob/9553f113879ed141a0a9ae74200c8cd02effa704/zerver/tests/test_subs.py#L3575) we directly call list_to_streams, so we need to add an error here for guest case which we do not do for `invite_to_stream_policy`. I am not sure why we need to call `list_to_streams` and assert the error [here](https://github.com/zulip/zulip/blob/9553f113879ed141a0a9ae74200c8cd02effa704/zerver/tests/test_subs.py#L3575). I would want suggestions here on how to handle this.

Reply:
I don't think it's important that we have any tests calling `list_to_streams` directly; my guess is that these were written to get coverage early on and would have been better written as end-to-end API tests.